integrate/simpsons_test.go

```diff
@@ -21,20 +21,24 @@ func TestSimpsons(t *testing.T) {
 		tol      float64
 	}{
 		{integral: testquad.Constant(0), n: 3, tol: 0},
+		{integral: testquad.Constant(0), n: 4, tol: 0},
 		{integral: testquad.Constant(0), n: 10, tol: 0},
 		{integral: testquad.Poly(0), n: 3, tol: 1e-14},
+		{integral: testquad.Poly(0), n: 4, tol: 1e-14},
 		{integral: testquad.Poly(0), n: 10, tol: 1e-14},
 		{integral: testquad.Poly(1), n: 3, tol: 1e-14},
+		{integral: testquad.Poly(1), n: 4, tol: 1e-14},
 		{integral: testquad.Poly(1), n: 10, tol: 1e-14},
 		{integral: testquad.Poly(2), n: 3, tol: 1e-14},
+		{integral: testquad.Poly(2), n: 4, tol: 1e-14},
 		{integral: testquad.Poly(2), n: 10, tol: 1e-14},
-		{integral: testquad.Poly(3), n: 1e3, tol: 1e-8},
-		{integral: testquad.Poly(4), n: 1e3, tol: 1e-8},
-		{integral: testquad.Poly(5), n: 1e3, tol: 1e-7},
-		{integral: testquad.Sin(), n: 1e2, tol: 1e-8},
-		{integral: testquad.XExpMinusX(), n: 1e2, tol: 1e-8},
+		{integral: testquad.Poly(3), n: 1000, tol: 1e-8},
+		{integral: testquad.Poly(4), n: 1000, tol: 1e-8},
+		{integral: testquad.Poly(5), n: 1000, tol: 1e-7},
+		{integral: testquad.Sin(), n: 100, tol: 1e-8},
+		{integral: testquad.XExpMinusX(), n: 101, tol: 1e-8},
```

Comment:
This test was failing in my virtual machine, plus there will be a test case with a large odd number of nodes.

Reply:
Also failing on travis and appveyor:
```
--- FAIL: TestSimpsons (0.00s)
    simpsons_test.go:58: Test #16: ∫_0^1 x*exp(-x)dx, n=101: unexpected result; got=0.26424112862624427 want=0.26424111765711533 diff=1.096912893849833e-08
```

Reply:
Yes, I noticed it too late. Let's see if it passes now.